### PR TITLE
update openshift install docs and example

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Clarify Operator install docs in Openshift

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

n/a

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/install-openshift.md [hunk 1]
```diff
@@ -16,33 +16,40 @@ Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from Oper
...
 ![Datadog Operator in the OperatorHub](assets/operatorhub.png)
 
+Installation includes the creation of a `ServiceAccount` called `datadog-agent-scc` that is bound to two default OpenShift `SecurityContextConstraints`.
+
 3. Specify the namespace to install the Datadog Operator in, you can use `datadog` if you previously created the project or an existing one, such as `openshift-operators`:
 
 ![Deploy the operator in the datadog namespace](assets/datadognamespace.png)
 
-## Deploy the Datadog Agents with the operator
+**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] going forward.
```
Comment (drichards-87) [inline docs/install-openshift.md:25]:
```suggestion
**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] in versions later than 1.0.
```

File: docs/install-openshift.md [hunk 2]
```diff
@@ -68,15 +73,23 @@ The following contains the simplest configuration using the v2alpha1 object:
...
-3. Deploy the Datadog Agent with the configuration file above:
+Setting the `serviceAccountName` in the `nodeAgent` and `clusterAgent` `override` section ensures that these pods are associated with the necessary `SecurityContextConstraints` and RBACs.
+
+3. Apply the Datadog Agent manifest:
    ```shell
    oc apply -f path/to/your/datadog-agent.yaml
    ```
 
-The Datadog Agent runs with the `datadog-agent-scc` service account created by the Datadog Operator and is connected to the right SCCs.
+The Datadog Agent and Cluster Agent should now be running and collecting data to be viewed and alerted on in the Datadog web app.
```
Comment (drichards-87) [inline docs/install-openshift.md:88]:
```suggestion
The Datadog Agent and Cluster Agent should now be running and collecting data. This data can be viewed and alerted on in Datadog.
```

File: docs/install-openshift.md [hunk 3]
```diff
@@ -16,33 +16,40 @@ Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from Oper
...
-1. Create a Kubernetes secret with your API and APP keys:
+
+1. Create a Kubernetes secret with your API and App keys:
 
    ```shell
-   kubectl create secret generic datadog-secret -n datadog --from-literal api-key=<DATADOG_API_KEY> --from-literal app-key=<DATADOG_APP_KEY>
+   oc create secret generic datadog-secret -n datadog --from-literal api-key=<DATADOG_API_KEY> --from-literal app-key=<DATADOG_APP_KEY>
    ```
-   Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API and application keys][3]
+   Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API][5] and [Application keys][6]
```
Comment (drichards-87) [inline docs/install-openshift.md:40]:
```suggestion
   Replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your [Datadog API][5] and [Application keys][6].
```

File: docs/install-openshift.md [hunk 4]
```diff
@@ -16,33 +16,40 @@ Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from Oper
...
 3. Specify the namespace to install the Datadog Operator in, you can use `datadog` if you previously created the project or an existing one, such as `openshift-operators`:
 
 ![Deploy the operator in the datadog namespace](assets/datadognamespace.png)
 
-## Deploy the Datadog Agents with the operator
+**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] going forward.
+
+## Deploy the Datadog Agent with the Operator
+
+After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. It is recommended to use the Cluster Agent to manage cluster-level monitoring, and this will also be deployed by default.
```
Comment (drichards-87) [inline docs/install-openshift.md:29]:
```suggestion
After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. Datadog recommends that you use the Cluster Agent to manage cluster-level monitoring, which will automatically be deployed by default.
```

File: docs/install-openshift.md [hunk 5]
```diff
@@ -16,33 +16,40 @@ Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from Oper
...
+
 
-After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers the Datadog Agent's deployment in your OpenShift cluster. With this resource, the Agent deploys as a `DaemonSet` on every `Node` of your cluster.
+**Note**: In Datadog Operator versions `1.1.0` and above, the Conversion Webhook is **disabled** by default, and can be enabled with the command argument `--webhookEnabled`. In version `1.0.3` of the Datadog Operator, listing the Conversion Webhook is **enabled** by default. The conversion allows a smooth transition from the `v1alpha1` (deprecated) `DatadogAgent` CRD to `v2alpha1`.
```
Comment (drichards-87) [inline docs/install-openshift.md:32]:
```suggestion
**Notes**: 
- In Datadog Operator versions `1.1.0` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`. 
- In Datadog Operator version `1.0.3`, the conversion webhook is **enabled** by default. The conversion allows a smooth transition from the `v1alpha1` (deprecated) `DatadogAgent` CRD to `v2alpha1`.
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Left you some minor style-related feedback from the Docs Team.

File: examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml [hunk 6]
```diff
@@ -18,13 +18,19 @@ spec:
...
+        runAsUser: 0
         seLinuxOptions:
           user: system_u
           role: system_r
           type: spc_t
           level: s0
-        runAsUser: 0
+    clusterAgent:
+      serviceAccountName: datadog-agent-scc
+      replicas: 2
```
Comment (clamoriniere) [inline examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml:36]:
```suggestion
      replicas: 2

```

File: examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml [hunk 7]
```diff
@@ -18,13 +18,19 @@ spec:
       apiKey: <DATADOG_API_KEY>
       appKey: <DATADOG_API_KEY>
     criSocketPath: /var/run/crio/crio.sock
+    kubelet:
+      tlsVerify: false
```
Comment (clamoriniere) [inline examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml:22]: is it mandatory?

File: docs/install-openshift.md [hunk 8]
```diff
@@ -16,33 +16,42 @@ Use the [Operator Lifecycle Manager][2] to deploy the Datadog Operator from Oper
...
+**Note**: Prior to version 1.0, multiple `InstallModes` were supported in the `ClusterServiceVersion` (see the [OLM operator install doc][3] as a reference). Due to the introduction of the conversion webhook in 1.0, only the `AllNamespaces` `InstallMode` [is supported][4] in versions 1.0 and later.
+
+## Deploy the Datadog Agent with the Operator
+
+After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers a deployment of the Datadog Agent in your OpenShift cluster. The Agent is deployed as a `Daemonset`. Datadog recommends that you use the Cluster Agent to manage cluster-level monitoring, which will automatically be deployed by default.
+
 
-After deploying the Datadog Operator, create a `DatadogAgent` resource that triggers the Datadog Agent's deployment in your OpenShift cluster. With this resource, the Agent deploys as a `DaemonSet` on every `Node` of your cluster.
+**Notes**:
+- In Datadog Operator versions `1.1.0` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`.
```
Comment (celenechang) [inline docs/install-openshift.md:33]:
```suggestion
- In Datadog Operator versions `1.1` and later, the conversion webhook is **disabled** by default. To enable the webhook, use the command argument `--webhookEnabled`.
```

**Review by drichards-87 — APPROVED** [review]

File: examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml (hunk 7 above)
Comment (clamoriniere) [reply examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml:22]: I think another solution is to mount the openshift root certificate

Comment (clamoriniere) [reply examples/datadogagent/v2alpha1/datadog-agent-openshift.yaml:22]: can we add a comment to say that `tlsVerify: false` is needed of the kubelet certificate is a self-signed certificat. 

and another way is to mount the CA certificat that was used to create the kubelet certificate.

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**